tests/task.test.ts

```diff
@@ -0,0 +1,29 @@
+import { taskReducer, TaskState, setCurrentTaskFromChannel} from '../src/task';
+import { Channel } from 'twilio-chat/lib/channel';
+
+jest.mock('../src/task', () => ({
+  setCurrentTaskFromChannel: jest.fn()
+}))
+
+describe('task reducer', () => {
+  test('should return initial state when no action is passed', () => {
+    const expected = {};
+    const initialState = undefined;
+    // const channel: Partial<Channel> = {attributes: { taskSid: null } }
+    const action = setCurrentTaskFromChannel(null)
+    const updatedState = taskReducer(initialState, action);
+
+    expect(updatedState).toStrictEqual(expected);
+  });
+
+  test('should return new taskSid when updating channel with current task', () => {
+    const initialState: TaskState = { currentSid: 'Task1' };
+    const newSid = 'Task2';
+    const expected = { ...initialState, currentSid: newSid };
+    const channel: Partial<Channel> = {attributes: { taskSid: newSid } }
+    const action = setCurrentTaskFromChannel(channel);
+    const updatedState = taskReducer(initialState, action);
```

Comment:
I think I am missing something for line 23. It seems it is necessary to define the channel with all the properties which are not relevant for this test. 
I get the following error for 'channel': 
```Argument of type 'Partial<Channel>' is not assignable to parameter of type 'Channel'.
  Type 'Partial<Channel>' is missing the following properties from type 'Channel': services, channelState, statusSource, entityPromise, and 6 more.ts(2345)```

Reply:
I wonder what you would do in this situation.

Reply:
Just use a cast, e.g. `channel as Channel` - casting should be kept to a minimum in prod code, but can be used more liberally in test setup code., otherwise you will end up drowning in unnecessary boilerplate, as this situation showss